### ISSUE TITLE
languageServices.getDocumentHighlights not returning results for usage in import

### ISSUE THREAD

**Issue by HamletDRC** [palantir/tslint#591]
I am finding some issues in the languageServices exposed from TypeScript exposed through TSLint. 

My rule is called "no unused import rule".
Here is my sample code that should not trigger a violation: 

       import DM = require("DM");
       import AB = DM.Dependency;
       console.log(AB)

The DM identifier is referenced, the AB identifier is referenced, and AB is used. This should not be an "unused import". 

To detect an unused import I call languageServices.getDocumentHighlights. 
You can see my usage in line 31 of my rule:
* https://github.com/Microsoft/tslint-microsoft-contrib/blob/master/src/noUnusedImportsRule.ts

I also have a unit test that displays the issue here: 
* https://github.com/Microsoft/tslint-microsoft-contrib/blob/master/tests/NoUnusedImportsTests.ts
* see test called "FALSE POSITIVE"

If this is too vague please let me know, I can give more details. 

Some possibility for the issue: 
* I am not creating the languageServices instance correctly
* there is a bug in the way TSLint instantiates/exposes the languageServices
* there is a bug in the TypeScript language services

Any help would be appreciated. Thanks.

**Comment by HamletDRC** [comment]
Ack... I am sorry, I just saw my mistake. Sorry to waste your time :(

**Comment by adidahiya** [comment]
TIL about tslint-microsoft-contrib. That's pretty cool. I think we'd be happy to accept some of these rules in the core.

btw the `no-unused-variable` rule does check for unused imports, but it could use some enhancements in terms of its failure message: https://github.com/palantir/tslint/issues/550

**Comment by HamletDRC** [comment]
@adidahiya Our team created tslint-microsoft-contrib for the Microsoft //oneweek Hackathon in July. I would love to see some of these rules in TSLint as well. As long as you do not violate our MIT license then I think you're free to take whatever you'd like into your project. I tried to write the unit tests in a similar manner to yours so that they are easy to move. The reason we did not just write the rules in your product and then submit pull requests to you is that it is a little easier to get OSS contribution approval for a project we control versus one a 3rd party controls. I have submitted the paperwork though, and I hope in the next few weeks to have permission to send you a pull request. 

We have a few issues on our side to fix before making our rules a "1.0" release. At some point I had planned to reach out to your project team and ask for a quick review/quality check. I'd still do that, I just still have some issues on my side to fix. Thanks!
